qa/rpc-tests/p2p-autoinstantsend.py

```diff
@@ -0,0 +1,289 @@
+                                         ['-debug=masternode', '-externalip=127.0.0.1',
+                                          '-masternode=1',
+                                          '-masternodeprivkey=%s' % self.mninfo[idx].key
+                                          ]))
+            for i in range(0, idx + 1):
+                connect_nodes(self.nodes[i], idx + 1)
+
+    def sentinel(self):
+        for i in range(1, self.mn_count + 1):
+            self.nodes[i].sentinelping("1.1.0")
+
+    def setup_network(self):
+        self.nodes = []
+        # create faucet node for collateral and transactions
+        self.nodes.append(start_node(0, self.options.tmpdir, ["-debug"]))
+        required_balance = MASTERNODE_COLLATERAL * self.mn_count + 1
+        while self.nodes[0].getbalance() < required_balance:
+            set_mocktime(get_mocktime() + 1)
+            set_node_times(self.nodes, get_mocktime())
+            self.nodes[0].generate(1)
+        # create masternodes
+        self.prepare_masternodes()
+        self.write_mn_config()
+        stop_node(self.nodes[0], 0)
+        self.nodes[0] = start_node(0, self.options.tmpdir,
+                                   ["-debug", "-sporkkey=cP4EKFyJsHT39LDqgdcB43Y3YXjNyjb5Fuas1GQSeAtjnZWmZEQK"])
+        self.create_masternodes()
+        # create connected simple nodes
+        for i in range(0, self.num_nodes - self.mn_count - 1):
+            self.create_simple_node()
+        # feed the sender with some balance
+        sender_addr = self.nodes[self.sender_idx].getnewaddress()
+        self.nodes[0].sendtoaddress(sender_addr, 1)
+        # make sender funds mature for InstantSend
+        for i in range(0, 2):
+            set_mocktime(get_mocktime() + 1)
+            set_node_times(self.nodes, get_mocktime())
+            self.nodes[0].generate(1)
+        # sync nodes
+        self.sync_all()
+        set_mocktime(get_mocktime() + 1)
+        set_node_times(self.nodes, get_mocktime())
+        sync_masternodes(self.nodes)
+        for i in range(1, self.mn_count + 1):
+            res = self.nodes[0].masternode("start-alias", "mn%d" % i)
+            assert(res["result"] == 'successful')
+        sync_masternodes(self.nodes)
+        #self.sentinel()
+        mn_info = self.nodes[0].masternodelist("status")
+        assert(len(mn_info) == self.mn_count)
+        for status in mn_info.values():
+            assert(status == 'ENABLED')
+
+    def get_autoix_bip9_status(self):
+        info = self.nodes[0].getblockchaininfo()
+        return info['bip9_softforks']['autoix']['status']
+
+    def activate_autoix_bip9(self):
+        # sync nodes periodically
+        # if we sync them too often, activation takes too manty time
+        # if we sync them too rarely, nodes failed to update its state and
+        # bip9 status is not updated
+        # so, in this code nodes are synced once per 20 blocks
+        counter = 0
+        sync_period = 20
```

Comment:
I noticed that 20 is too much and nodes fail to sync mnw sometimes which results in a fork/rejected blocks. I think 10 should be fine (because we determine winners for 10 blocks ahead). Well, at least I don't have the issue locally anymore after changing it this way.